### PR TITLE
Cleanup CI configuration

### PR DESCRIPTION
**Pull request by Turbo87**
see commit messages, best reviewed commit-by-commit

/cc @rwjblue

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -63,11 +34,34 @@ env:
...
 
+stages:
+  - basic test
+  - additional tests
+  - name: deploy
+    if: type IN (push)
+
+before_install:
+  - export DISPLAY=:99.0
+  - sh -e /etc/init.d/xvfb start
```
Comment (stefanpenner) [inline .travis.yml]: Are these two lines still needed?

```js
  - export DISPLAY=:99.0
  - sh -e /etc/init.d/xvfb start
```

I don't believe we still require anything running in a virtual frame buffer. I suspect old browser testing needed this.

Comment (rwjblue) [reply .travis.yml]: I suspect they are not needed any more

Comment (Turbo87) [reply .travis.yml]: I was assuming that maybe Firefox still needed that, so I left it in there for now.

**Review by rwjblue — APPROVED** [review]

File: .travis.yml (hunk 1 above)
Comment (Turbo87) [reply .travis.yml]: looks like this is indeed still needed (see https://travis-ci.org/emberjs/ember.js/jobs/463512118)

I've dropped the commit that removes it again. I suggest merging this and then we can iterate forward.

Comment (stefanpenner) [reply .travis.yml]: Oh, I didn't realize we had a smoke test for FF usage using xvfb. Future work could likely just use https://developer.mozilla.org/en-US/docs/Mozilla/Firefox/Headless_mode and avoid that. 

But that shouldn't hold up this PR, thanks for investigating.

**Review by stefanpenner — APPROVED** [review]

**Pull request merged**